server/graphql/v1/queries.js

```diff
@@ -327,14 +343,57 @@ const queries = {
       if (!collective) throw new Error('This collective does not exist');
 
       // Load transactions
-      return collective.getTransactions({
+      const allTransactions = await collective.getTransactions({
+        attributes: attributes,
         order: [['createdAt', 'DESC']],
         type: args.type,
         limit: args.limit,
         offset: args.offset,
         startDate: args.dateFrom,
         endDate: args.dateTo,
       });
+      if (!fetchDataFromLedger) return allTransactions;
+
+      const ledgerQuery = {
+        where: {
+          ToAccountId: args.CollectiveId,
+        },
+        limit: args.limit,
+        offset: args.offset,
+      };
+      ledgerQuery.where = JSON.stringify(ledgerQuery.where);
+      const transactionsEndpointResult = await axios.get(
+        `${config.ledger.transactionUrl}?${queryString.stringify(ledgerQuery)}`,
+      );
+      const ledgerTransactions = groupBy(transactionsEndpointResult.data || [], 'LegacyCreditTransactionId');
+      // sort keys of result by legacy id DESC as lodash groupBy changes the order
+      const ledgerFormattedTransactions = [];
+      for (const key of Object.keys(ledgerTransactions).sort((a, b) => b - a)) {
+        // mapping legacy info to parse inside ledger mapping
+        const legacyMatchingTransaction = allTransactions.filter(t => {
+          return (
+            t.id === ledgerTransactions[key][0].LegacyDebitTransactionId ||
+            t.id === ledgerTransactions[key][0].LegacyCreditTransactionId
+          );
+        })[0];
+        let uuid, VirtualCardCollectiveId, HostCollectiveId, RefundTransactionId;
+        if (legacyMatchingTransaction) {
+          uuid = legacyMatchingTransaction.uuid;
+          VirtualCardCollectiveId = legacyMatchingTransaction.UsingVirtualCardFromCollectiveId;
+          HostCollectiveId = legacyMatchingTransaction.HostCollectiveId;
+          RefundTransactionId = legacyMatchingTransaction.RefundTransactionId;
+        }
+        ledgerFormattedTransactions.push(
+          parseLedgerTransactionToApiFormat(key, ledgerTransactions[key], {
+            AccountId: args.CollectiveId,
+            legacyUuid: uuid,
+            VirtualCardCollectiveId,
+            HostCollectiveId,
+            RefundTransactionId,
+          }),
+        );
+      }
+      return ledgerFormattedTransactions;
```

Comment:
Extract all this block in a separated `ledger` file, which should expose a `getTransactions` method or similar.